### DISCUSSION TITLE
What is the best way to use mycollate?

### DISCUSSION THREAD

**Discussion by Fei-Wang** [open-mmlab/mmengine#899] [Q&A] [answered]
below is my code
```python3
from mmengine.dataset import COLLATE_FUNCTIONS as MMENGINE_COLLATE_FUNCTIONS
COLLATE_FUNCTIONS = Registry(
    'Collate Functions',
    parent=MMENGINE_COLLATE_FUNCTIONS,
    locations=['myproject.datasets'],
)
```
and 
```python3
@COLLATE_FUNCTIONS.register_module()
def mycollate(data_batch, *args, **kwargs):
    pass
```
and in myconfig.py
```python3
collate = dict(
    type='mycollate',
)
train_dataloader=dict(collate=collate,...)
```
but, default runner will report error at 
https://github.com/open-mmlab/mmengine/blob/5753cd98e0b791897a2d402b7eb196fb04b4a4eb/mmengine/runner/runner.py#L1402
so, what is the best way to use mycollate?

**Comment by C1rN09** [comment, marked as answer]
There is an issue with `Registry.get()` method that **it will not infer scopes** as `Registry.build()` do. As a result, you have to register your `mycollate` to MMEngine's own root registry rather than your child registry.

Honestly speaking, this behavior is kind of weird. We will try to find a better solution in the future.
